models/common data models/recharge__line_item_enhanced.sql

```diff
@@ -0,0 +1,202 @@
+with charge_line_items as (
+
+    select * 
+    from {{ var('charge_line_item')}}
+
+), charges as (
+
+    select * 
+    from {{ var('charge') }}
+
+), charge_shipping_lines as (
+
+    select
+        charge_id,
+        sum(price) as total_shipping
+    from {{ var('charge_shipping_line') }}
+    group by 1
+
+{% if var('recharge__using_checkout', false) %}
+), checkouts as (
+
+    select *
+    from {{ var('checkout') }}
+
+{% endif %}
+
+), addresses as (
+
+    select * 
+    from {{ var('address') }}
+
+), customers as (
+
+    select * 
+    from {{ var('customer') }}
+
+), subscriptions as (
+
+    select *
+    from {{ var('subscription_history') }} 
+    where is_most_recent_record
+
+), enhanced as (
+    select
+        charge_line_items.charge_id as header_id,
+        charge_line_items.index as line_item_id,
+        row_number() over (partition by charge_line_items.charge_id
+            order by charge_line_items.index) as line_item_index,
+
+        -- header level items
+        charges.charge_created_at as created_at,
+        charges.charge_status as header_status,
+        charges.total_discounts as discount_amount,
+        charges.total_refunds as refund_amount,
+        charge_shipping_lines.total_shipping as fee_amount,
+        addresses.payment_method_id,
+        charges.external_transaction_id_payment_processor as payment_id,
+        charges.payment_processor as payment_method,
+        charges.charge_processed_at as payment_at,
+        charges.charge_type as billing_type,  -- possible values: checkout, recurring
+
+        {% if var('recharge__using_checkout', false) %}
+        checkouts.currency,
+        {% else %}
+        cast(null as {{ dbt.type_string() }}) as currency, 
+        -- currency is in the charges api but not the fivetran schema, so relying on checkouts for now. this only has 20% utilization though so we will want to switch if they add it.
+        {% endif %}
+
+        charge_line_items.purchase_item_type as transaction_type, -- possible values: subscription, onetime
```

Comment:
I considered just making this line `'charge' as transaction_type` since everything in this table is a charge and would look more like the other packages, but I thought this was more useful for Recharge specifically. For example, a charge might be generally recurring, but it might have a onetime item that a customer added on. Open to

Reply:
I agree with your decision to have this be `purchase_item_type`. This fits better for the platform as whole and tells a more complete story.